### PR TITLE
syncer: fix issues when switching to core.Consensus

### PR DESCRIPTION
**Pull request by missionliao**
- when confirmed blocks passed to `core.Consensus` aren't continuous in position in some chain, the pulling would skip those missing blocks.
   - fix: when some block is missing, avoid adding it and all blocks after it to `core.Consensus`.
- we need to avoid the receive channel of network module full
  - fix: during switching to `core.Consensus`, we need to launch a dummy receiver to receive from receive channel of network module.
  - fix: between the period during `core.Consensus` created and before running, a dummy receiver is also required to receive from receive channel of network module.

### CODE REVIEW

File: core/consensus.go [hunk 1]
```diff
@@ -596,12 +622,25 @@ func (con *Consensus) Run() {
...
+	// Stop dummy receiver if launched.
+	if con.dummyCancel != nil {
+		con.logger.Trace("Stop dummy receiver")
+		con.dummyCancel()
+		<-con.dummyFinished
+		// Replay those cached messages.
+		con.logger.Trace("Dummy receiver stoped, start dumping cached messages",
+			"count", len(con.dummyMsgBuffer))
+		for _, msg := range con.dummyMsgBuffer {
+			con.handleMsg(msg)
```
Comment (jimmyhu-dexon) [inline core/consensus.go]: `handleMsg` will be called by two go routine and this function might not be thread safe

Comment (missionliao) [reply core/consensus.go]: Fixed in 3803d50

**Comment by missionliao** [comment]
@jimmyhu-dexon @haoping-ku  fixed an issue that while some blocks in confirmed block set acking others in the same set, they won't be added to lattice.

**Comment by missionliao** [comment]
@jimmyhu-dexon @haoping-ku fix an issue that the stopped node in `TestConsensus/TestSync` stopped in round#4, instead of round#3.

**Pull request merged**